Pathwayz sidebar

ZeeJab opened the pull request:
Cleans up code from PR #247 for clearer understanding of what is happening:

More specifically it follows OOCSS conventions for cleaner files && a better understanding of what is going on for anyone that wants to debug later

Interesting fact: 
buttons files went from `160 lines to 98 o.O`
pathway-editor went from `45 lines to 26 -.o`
and html is much `cleaner` now

I was not able to refactor the js much but i did move the event handlers in editor.js on top of initialize as per backbone conventions. 

Once this PR is merged it adds the add, delete, undo, etc sidebar to the pathway pledge page as per #269

Commit:
added sidebar buttons on pledged pathway page

Commit:
removed some commented lines

Commit:
added the pathway add-badge functionality

Commit:
moved pathway editor buttons to button.scss

Commit:
removed re-add of starbucks logo

Commit:
refactored editor.js

Commit:
initiated side label tag for add badge on pathway editor sidebar

Commit:
added sidebar hover tags and stylized

ZeeJab committed:
Cleans up html/css for sidebar menu #269

follows OOCSS convensions for cleaner file && 
a better understanding of what is going on for
anyone that wants to debug

buttons files went from 160 line to 98 o.O
pathway-editor went from 45 lines to 26 -.o
and html is much cleaner now

ZeeJab committed:
Had a hard time refactoring this js file #269

but i moved the event handlers above initialize

ZeeJab committed:
rebased with master

stenington commented:
http://discovery-285.herokuapp.com/

chloeatplay commented:
Quick q: why do the badge squares look so small? I am assuming they all look like they are on one column because @LucasBlair @ghami need to define positions, right?

ZeeJab commented:
@chloeatplay I assumed it was because of `<div class="small-12 medium-8 medium-centered columns">` that wraps the canvas. This already exists in master so I did not remove it. @stenington?

chloeatplay commented:
Hmm there are a bunch of things not quite looking as they should here
a) before pledging all the badges look tiny and in one straight column
![discover open badges 2014-05-12 10-15-48](https://cloud.githubusercontent.com/assets/1945209/2945367/449dbbd6-d9e0-11e3-8bf0-a1a9ea9131fb.png)

b) after pledging the badges look bigger but their title text doesn't wrap
![discover open badges 2014-05-12 10-16-05](https://cloud.githubusercontent.com/assets/1945209/2945369/4acc5be8-d9e0-11e3-9b59-26315913db83.png)
c) there are some funky things going on with the buttons, where their status remains blue after the interactions. In addition we will not need the "recommended" buttons for sure, and the "add notes" one is TBD i guess. 
d) the add badges section still does the dancing squares thingie, also those squares look huge @zzzndr whats the size for those? 
![discover open badges 2014-05-12 10-16-33](https://cloud.githubusercontent.com/assets/1945209/2945377/682f82b4-d9e0-11e3-9fca-c4fa89ddd153.png)

ZeeJab commented:
Okii, will look into it. But also keep in mind, the only thing that I've cleaned up in this PR is the actual sidebar and how it looks...nothing else has been worked on yet

zzzndr commented:
for desktop the intended size for the badge containers is 210x210 pixels (:

chloeatplay commented:
@ZeeJab okayz!

stenington commented:
I'll fix the add panel in another PR, I have some mostly completed work that cleans all of that up that I can make a PR for after this one goes in.

stenington commented:
That commit takes care of the pathway being overly narrow when not pledged.

stenington commented on clientapp/styles/add-badge-button.scss:
```diff
@@ -0,0 +1,37 @@
+.achievement .token {
```
This file seems weird given its name and contents. Any idea what happened @ZeeJab?

stenington replied:
Oh, I guess it's styling achievements when they have a little checkbox for being added to pathways so it makes sense.

ZeeJab replied:
yea :-( im cleaning this up now too

stenington commented:
Styling of badges on a pathway is handled very differently than badges appearing anywhere else because it's done in canvas. Even though it looks wrong, we're probably better served by merging this once the buttons work and dealing with styling those separately.

chloeatplay commented:
@stenington kk, makes sense :thumbsup:

ZeeJab committed:
get's rid of duplicate styling #285

ZeeJab committed:
makes .full-length-hr universal for styles #285

ZeeJab committed:
fixes padding and adds border radiums #285

for body-pathway
adds padding for pathway-page and landing-page
gets rid of extra space in stylesheet

ZeeJab committed:
adds correct styling for settings icon #285

ZeeJab committed:
deletes extra styling #285

ZeeJab committed:
adds clearfix to styles and adds full-length-hr to all styles #285

ZeeJab committed:
styles hr tags and view more button for adding badge page #285

ZeeJab committed:
fixes weird spacing issues in file #285

and gets rid of unnecessary <i> tag

stenington committed:
Sizing pathway editor to use full width for display

ZeeJab committed:
changes color and shape for delete button when 
you want to delete a badge #285

ZeeJab committed:
Merge branch 'pathwayz-sidebar' of github.com:mozilla/openbadges-discovery into pathwayz-sidebar

stenington committed:
Make pathway editor buttons toggle properly

stenington commented:
I pushed another version out to http://discovery-285.herokuapp.com/

stenington commented:
Hmm, the "new" indicators are now way off the mark.

stenington commented:
This PR has maybe gotten a little out of hand. I think I'm actually going to cherry-pick from it and issue many small PRs.

ZeeJab commented:
@stenington we can close this now right?

stenington commented:
yeppers
